packages/node_modules/@ciscospark/react-component-adaptive-card/src/index.js

```diff
@@ -17,11 +21,53 @@ class AdaptiveCard extends React.Component {
   componentDidMount() {
     if (Object.prototype.hasOwnProperty.call(this, 'nodeElement') && Object.prototype.hasOwnProperty.call(this.nodeElement, 'current')) {
       this.nodeElement.current.appendChild(
-        getAdaptiveCard(this.props.cards, this.props.displayName, this.props.sparkInstance, this.addChildNode)
+        getAdaptiveCard(this.state.cards, this.props.displayName, this.props.sparkInstance, this.addChildNode)
       );
     }
   }
 
+  static getDerivedStateFromProps(nextProps) {
+    if (nextProps.verb === API_ACTIVITY_VERB.SHARE) {
+      try {
+        const decryptedURLs = nextProps.items
```

Comment:
Just trying to understand the code here, where does `verb` and `items` come from?
My understanding is that `nextProps` represent the component's props, but from where i can see those are only `cards`, `displayname` and `sparkInstance`

(side note - we shouldn't propagate the name "spark" and we should use "webex")

Reply:
[Adaptive card component](https://github.com/webex/react-widgets/blob/5b1a635ef2b1e5eb25e53cf1b45ef0330dcb3df6/packages/node_modules/%40ciscospark/react-component-adaptive-card/src/index.js) calling from two places.One [component](https://github.com/webex/react-widgets/blob/5b1a635ef2b1e5eb25e53cf1b45ef0330dcb3df6/packages/node_modules/%40ciscospark/react-component-activity-item/src/index.js)  is passing three props and
 [share component](https://github.com/webex/react-widgets/blob/5b1a635ef2b1e5eb25e53cf1b45ef0330dcb3df6/packages/node_modules/%40ciscospark/react-component-activity-share-files/src/index.js) need to pass more props. In my use case, I'm getting the required props from a [share component](https://github.com/webex/react-widgets/blob/5b1a635ef2b1e5eb25e53cf1b45ef0330dcb3df6/packages/node_modules/%40ciscospark/react-component-activity-share-files/src/index.js) as `nextProps`. Though those are defined  in PropTypes it will throw an error because `this.props` is not using anywhere for remaining `verb`, `items` and `share`.

Reply:
![Screenshot 2019-12-13 at 12 14 52 PM](https://user-images.githubusercontent.com/55378170/70776143-40f61d00-1da2-11ea-8774-ae82d79aa1f1.png)
Though default values for props are added, esLint is throwing an error . Can I disable the esLint here?

Reply:
added propTypes and defaultProps

Reply:
![Screenshot 2019-12-13 at 12 14 52 PM](https://user-images.githubusercontent.com/55378170/70791979-3fd2e900-1dbe-11ea-831f-66c17ffcaccb.png)
So can I add propTypes and disable esLint?

Reply:
added `componentDidUpdate` and added extra props in propTypes.I have to set state for re-render.